File: sdks/python/examples/create_swap.py
"""End-to-end example: create an XLM→BTC swap order with the Python SDK.

Run with:
    pip install chainbridge
    export CHAINBRIDGE_API_KEY=cb_xxx
    export CHAINBRIDGE_SENDER=GA...
    python create_swap.py
"""

from __future__ import annotations

import os
import sys

from chainbridge import ChainBridgeClient


def main() -> int:
    api_key = os.environ.get("CHAINBRIDGE_API_KEY")
    sender = os.environ.get("CHAINBRIDGE_SENDER")
    if not api_key or not sender:
        print("Set CHAINBRIDGE_API_KEY and CHAINBRIDGE_SENDER first.", file=sys.stderr)
        return 1

    with ChainBridgeClient(api_key=api_key) as client:
        fees = client.market.estimate_fees(
            from_chain="stellar", to_chain="bitcoin", from_amount="1000000000"
        )
        print(f"Estimated fees: {fees}")

        order, secret, hash_lock = client.create_swap_order(
            from_chain="stellar",
            to_chain="bitcoin",
            from_asset="XLM",
            to_asset="BTC",
            from_amount="1000000000",
            to_amount="10000",
            sender_address=sender,
            expiry_seconds=86_400,
        )
        print(f"Created order: {order.order_id}")
        print(f"Hash-lock (share with counterparty): {hash_lock}")
        print(f"Secret (KEEP PRIVATE until claim): {secret}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: sdks/rust/examples/create_swap.rs
//! Create an XLM→BTC swap order with the Rust SDK.
//!
//! Run with:
//!     CHAINBRIDGE_API_KEY=cb_xxx CHAINBRIDGE_SENDER=GA... \
//!     cargo run --example create_swap

use chainbridge_sdk::{
    crypto::{derive_hash_lock, generate_secret_default},
    ChainBridgeClient, ClientConfig, CreateOrderInput, FeeEstimateInput,
};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let api_key = std::env::var("CHAINBRIDGE_API_KEY")?;
    let sender = std::env::var("CHAINBRIDGE_SENDER")?;
    let base = std::env::var("CHAINBRIDGE_API_URL")
        .unwrap_or_else(|_| "https://api.chainbridge.io".to_string());

    let client = ChainBridgeClient::new(ClientConfig {
        base_url: base,
        api_key: Some(api_key),
        ..Default::default()
    })?;

    let fees = client
        .market()
        .estimate_fees(&FeeEstimateInput {
            from_chain: "stellar".into(),
            to_chain: "bitcoin".into(),
            from_amount: "1000000000".into(),
        })
        .await?;
    println!("Estimated fees: total ${} ({} bps protocol)", fees.total_fee_usd, fees.protocol_fee_bps);

    let secret = generate_secret_default();
    let hash_lock = derive_hash_lock(&secret)?;
    let order = client
        .orders()
        .create(&CreateOrderInput {
            from_chain: "stellar".into(),
            to_chain: "bitcoin".into(),
            from_asset: "XLM".into(),
            to_asset: "BTC".into(),
            from_amount: "1000000000".into(),
            to_amount: "10000".into(),
            sender_address: sender,
            expiry: 86_400,
        })
        .await?;

    println!("Created order: {}", order.order_id);
    println!("Hash-lock (share with counterparty): {hash_lock}");
    eprintln!("Secret (KEEP PRIVATE until claim): {secret}");
    Ok(())
}
